### DISCUSSION TITLE
wait a go method that returns nothing?

### DISCUSSION THREAD

**Discussion by kybin** [wailsapp/wails#1437] [Q&A] [answered]
I found it does not create a Promise from go methods that doesn't return anything.

I wrote an App method like this.
```
func (a *App) GoTo(path string) { ... }
```

When I tried to get a Promise from the function call,
```
GoTo(path).then(function() { redraw(); })
```

It raises an error.
```
[Error] TypeError: undefined is not an object (evaluating 'GoTo(path).then')
	(anonymous function) (main.js:6)
```

Is there another way to wait until the call has done?
I know I can fix the situation by change its signature to return string (or anything).
```
func (a *App) GoTo(path string) string { ... }
```
Which I'm not convinced as the returned string isn't meaningful.

Thank you.

**Comment by ianmjones** [comment, marked as answer]
There's a PR that re-introduces a `return` in the generated JavaScript so that you can `await` the result if you wish.

https://github.com/wailsapp/wails/pull/1410

Lea is waiting on some feedback from people about this and possibly changing the way events work too. It would be great if you could share your thoughts there.

**Reply by leaanthony** [reply]
Yeah, I think the PR is good to go if that solves your problem @kybin

**Reply by kybin** [reply]
I don't understand the conversation entirely (I'm not a javascript expert), and I'd like to have blocking function calls than Promises. But there might be a reason for that I guess.

The change will definitely help my problem. Thanks.

**Reply by leaanthony** [reply]
Merged. Please let me know how you get on 👍

**Reply by kybin** [reply]
@leaanthony 

Hm.. I got the latest version and ran my app again, it doesn't raise error but it seems like there's race condition.

And I am seeing this in my `frontend/wailsjs/go/main/App.js`

```
export function GoTo(arg1) {
  window['go']['main']['App']['GoTo'](arg1);
}
```

Is it correct? Shouldn't it return?

I'm a bit confused, I might missed some steps?

(I'm using js without typescript if it matters.)

**Reply by kybin** [reply]
OK. It didn't raise error because I used `await` instead of `.then`

The generated code always same as before for me.

No matter what I do for re-install wails command, it always creates the same code.

I clone and modified the code but it was not reflecting.. it even builds without the generate.go file.

This is the step for building/breaking the app.

```
git clone https://github.com/wailsapp/wails
cd wails/v2/cmd/wails
rm ../../internal/binding/generate.go
go install
```

It ran successfully...

**Reply by ianmjones** [reply]
Did you update your project's `go.mod` to add a `replace` reference to the local checkout of the wails package?

https://wails.io/docs/next/guides/bleeding-edge

After doing that, best bet is to run `wails dev` so it generates the bindings again.

**Reply by kybin** [reply]
Oh thanks, I didn't knew that. I thought it only uses the command.

Now it generates the binding code properly. And there isn't an error nor race condition anymore.

Thank you! @ianmjones @leaanthony
